ironfish/src/rpc/routes/eth/getTransactionByHash.ts

```diff
@@ -0,0 +1,143 @@
+  yup
+    .object({
+      blockHash: yup.string().defined(),
+      blockNumber: yup.string().defined(),
+      from: yup.string().defined(),
+      gas: yup.string().defined(),
+      gasPrice: yup.string().defined(),
+      maxFeePerGas: yup.string().defined(),
+      maxPriorityFeePerGas: yup.string().defined(),
+      hash: yup.string().defined(),
+      input: yup.string().defined(),
+      nonce: yup.string().defined(),
+      to: yup.string().nullable().defined(),
+      transactionIndex: yup.string().defined(),
+      value: yup.string().defined(),
+      type: yup.string().defined(),
+      accessList: yup.array().of(yup.string().defined()).defined(),
+      chainId: yup.string().defined(),
+      v: yup.string().defined(),
+      r: yup.string().defined(),
+      s: yup.string().defined(),
+      yParity: yup.string().defined(),
+    })
+    .defined()
+
+registerEthRoute<typeof GetTransactionByHashRequestSchema, GetTransactionByHashResponse>(
+  `eth_getTransactionByHash`,
+  `${ApiNamespace.eth}/getTransactionByHash`,
+  GetTransactionByHashRequestSchema,
+  async (request, node): Promise<void> => {
+    Assert.isInstanceOf(node, FullNode)
+    const transactionHash =
+      await node.chain.blockchainDb.getEthTransactionHashToTransactionHash(
+        Buffer.from(EthUtils.remove0x(request.data), 'hex'),
+      )
+    if (!transactionHash) {
+      throw new RpcNotFoundError(`Transaction ${request.data} not found`)
+    }
+    const blockHash = await node.chain.getBlockHashByTransactionHash(transactionHash)
+
+    if (!blockHash) {
+      throw new RpcNotFoundError(`Block not found for transaction ${request.data}`)
+    }
+    const blockHeader = await node.chain.getHeader(blockHash)
+    if (!blockHeader) {
+      throw new RpcNotFoundError(`Block header not found for transaction ${request.data}`)
+    }
+    const transactions = await node.chain.getBlockTransactions(blockHeader)
+
+    let transaction:
+      | {
+          transaction: Transaction
+          initialNoteIndex: number
+          sequence: number
+          blockHash: Buffer
+          previousBlockHash: Buffer
+          timestamp: Date
+        }
+      | undefined = undefined
+    let transactionNum = 0
+    for (const tx of transactions) {
+      if (tx.transaction.hash().equals(transactionHash)) {
+        transaction = tx
+        break
+      }
+      transactionNum += 1
+    }
+    if (!transaction) {
+      throw new RpcNotFoundError(`Transaction ${request.data} not found`)
+    }
+    if (!transaction.transaction.evm) {
+      throw new RpcNotFoundError(`Transaction ${request.data} does not have EVM description`)
+    }
+    const ethTransaction = evmDescriptionToLegacyTransaction(transaction.transaction.evm)
+    // TODO deal with items that are mocked
+    request.end({
+      blockHash: EthUtils.prefix0x(transaction.blockHash.toString('hex')),
+      blockNumber: EthUtils.prefix0x(transaction.sequence.toString(16)),
```

Comment:
Could prob use a number / bigint to hex string util since we're using this a lot